### ISSUE TITLE
Generalize `expect` and `correlation_matrix` to MPOs

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#730]
Generalize `expect` and `correlation_matrix` to MPOs.

This comes up here: http://itensor.org/support/3350/calculating-observables-from-density-matrices and is needed in `PastaQ`.

A conceptual simple approach would be something like:
```julia
M::MPO
s = dag.(siteinds(first, M; plev=0))
x = "Sx"
O = op(x, s)
tr(apply([O], M))
```
where of course we would want to handle moving the orthogonality center around efficiently when computing ranges of local expectation values.

**Comment by emstoudenmire** [comment]
Yes, that forum question also made me realize we should have this.
